app/com/arpnetworking/metrics/portal/expressions/impl/DatabaseExpressionRepository.java

```diff
@@ -0,0 +1,303 @@
+        this(
+                ConfigurationHelper.<ExpressionQueryGenerator>getType(
+                        environment,
+                        config,
+                        "expressionRepository.expressionQueryGenerator.type")
+                        .newInstance());
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param expressionQueryGenerator Instance of <code>ExpressionQueryGenerator</code>.
+     */
+    public DatabaseExpressionRepository(final ExpressionQueryGenerator expressionQueryGenerator) {
+        _expressionQueryGenerator = expressionQueryGenerator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening expression repository").log();
+        _isOpen.set(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing expression repository").log();
+        _isOpen.set(false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Expression> get(final UUID identifier) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Getting expression")
+                .addData("expressionId", identifier)
+                .log();
+
+        final models.ebean.Expression ebeanExpression = Ebean.find(models.ebean.Expression.class)
+                .where()
+                .eq("uuid", identifier)
+                .findUnique();
+        if (ebeanExpression == null) {
+            return Optional.empty();
+        }
+
+        final Expression result = new DefaultExpression.Builder()
+                .setId(ebeanExpression.getUuid())
+                .setCluster(ebeanExpression.getCluster())
+                .setMetric(ebeanExpression.getMetric())
+                .setScript(ebeanExpression.getScript())
+                .setService(ebeanExpression.getService())
+                .build();
+        return Optional.of(result);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ExpressionQuery createQuery() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Preparing query").log();
+        return new DefaultExpressionQuery(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public QueryResult<Expression> query(final ExpressionQuery query) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Querying")
+                .addData("query", query)
+                .log();
+
+        // Create the base query
+        final PagedList<models.ebean.Expression> pagedExpressions = _expressionQueryGenerator.createExpressionQuery(query);
+
+        // Compute the etag
+        final SqlQuery sqlQuery = Ebean.createSqlQuery("SELECT CURRVAL('portal.expressions_etag_seq') AS etag;");
```

Comment:
Similar comments here as alerts db repository:
 - etag fetch to query generator
 - ebean to model conversion function
 - etag obfuscation
 - exceptions in error messages
 - exception handling